webapp/src/components/AssessmentMetadataBar.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  Avatar,
+  Divider,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Tooltip,
+  Typography,
+  Switch,
+} from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
+import InventoryIcon from '@mui/icons-material/Inventory';
+import TimerIcon from '@mui/icons-material/Timer';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import LightbulbIcon from '@mui/icons-material/Lightbulb';
+import { blue, grey } from '@mui/material/colors';
+
+import {
+  Assessment,
+  SubmissionStatus,
+  AssessmentType,
+  AssessmentSubmission,
+} from '../assets/data';
+import { formatDateTime } from '../helpers/dateTime';
+
+interface AssessmentMetadataBarProps {
+  assessment: Assessment;
+  submission: AssessmentSubmission;
+}
+
+const AssessmentMetadataBar = ({
+  assessment,
+  submission,
+}: AssessmentMetadataBarProps) => {
+  const dueTime = new Date(assessment?.dueDate!);
+  //TODO: use the opened date from the submission
+  const [openedTime] = useState(new Date());
+  const [secondsRemaining, setSecondsRemaining] = useState(
+    assessment?.testDuration! * 60
+  );
+
+  const requestRef = useRef<number>();
+  const previousTimeRef = useRef<number>();
+
+  const animate = (time: number) => {
+    if (previousTimeRef.current !== undefined) {
+      if (
+        new Date().getTime() > dueTime.getTime() ||
+        openedTime.getTime() + assessment?.testDuration! * 60 * 1000 <
+          new Date().getTime()
+      ) {
+        setSecondsRemaining(0);
+      } else if (
+        new Date().getTime() + assessment?.testDuration! * 60 * 1000 >
+        dueTime.getTime()
+      ) {
+        setSecondsRemaining(
+          Math.round((dueTime.getTime() - new Date().getTime()) / 1000)
+        );
+      } else {
+        setSecondsRemaining(
+          Math.round(
+            (openedTime.getTime() +
+              assessment?.testDuration! * 60 * 1000 -
+              new Date().getTime()) /
+              1000
+          )
+        );
+      }
+    }
+    previousTimeRef.current = time;
+    requestRef.current = requestAnimationFrame(animate);
```

Comment:
Perhaps we can wrap this in a test to only `requestAnimationFrame()` when secondsRemaining is greater than 0, or we could `cancelAnimationFrame` like we do in `useEffect()` below. No need to keep animating a value that's not going to change.